pyphare/pyphare/pharein/restarts.py

```diff
@@ -1,8 +1,75 @@
 import os
 import numpy as np
+from pathlib import Path
+
 from pyphare.core import phare_utilities
 
 
+def format_time(time):
+    """
+    0.006 == "00000.00600"
+    """
+    return "{:0>11.5f}".format(time)
+
+
+def dump(simulator, time, time_step):
+    if not simulator.cpp_sim.dump_restarts(timestamp=time, timestep=time_step):
+        return
+    # else a restart dump was made
```

Comment:
tell me if I get it correctly, this function dumps checkpoint if time is dump time AND it tries to delete some previous ones based on restart options.
I feel it could be clearer if everything below that "else" comment would be in a function like "remove_previous _checkpoints" or some shorter version of that